pkg/service/service.go

```diff
@@ -662,6 +706,108 @@ func (s *service) ListModels(ctx context.Context, pageSize int32, pageToken stri
 	return pbModels, int32(totalSize), nextPageToken, err
 }
 
+func (s *service) ListModelTriggers(ctx context.Context, req *modelpb.ListModelTriggersRequest, view modelpb.View) (*modelpb.ListModelTriggersResponse, error) {
+	pageSize := s.pageSizeInRange(req.GetPageSize())
+	page := s.pageInRange(req.GetPage())
+
+	orderBy, err := ordering.ParseOrderBy(req)
+	if err != nil {
+		return nil, err
+	}
+
+	var startTimeFrom, startTimeTo *time.Time
+	if req.StartTimeFrom.IsValid() {
+		t := req.GetStartTimeFrom().AsTime()
+		startTimeFrom = &t
+	}
+	if req.StartTimeTo.IsValid() {
+		t := req.GetStartTimeTo().AsTime()
+		startTimeTo = &t
+	}
+	logger, _ := custom_logger.GetZapLogger(ctx)
+
+	triggers, totalSize, err := s.repository.ListModelTriggers(ctx, int64(pageSize), int64(page), orderBy, req.ModelUid, startTimeFrom, startTimeTo)
+	if err != nil {
+		return nil, err
+	}
+
+	metadataMap := make(map[string][]byte)
+	if view == modelpb.View_VIEW_FULL {
+		var referenceIDs []string
+		for _, trigger := range triggers {
+			referenceIDs = append(referenceIDs, trigger.InputReferenceID)
+			if trigger.OutputReferenceID.Valid {
+				referenceIDs = append(referenceIDs, trigger.OutputReferenceID.String)
+			}
+		}
+
+		logger.Debug("start to get files from minio", zap.String("referenceIDs", strings.Join(referenceIDs, "/")))
+		fileContents, err := s.minioClient.GetFilesByPaths(ctx, referenceIDs)
+		if err != nil {
+			logger.Error("failed to get files from minio", zap.Error(err))
+			return nil, err
+		}
+		for _, content := range fileContents {
+			metadataMap[content.Name] = content.Content
+		}
+	}
+
+	pbTriggers := make([]*modelpb.ModelTrigger, len(triggers))
+	for i, trigger := range triggers {
+		pbTrigger := &modelpb.ModelTrigger{
+			Uid:          trigger.UID.String(),
+			ModelUid:     trigger.ModelUID.String(),
+			TriggerUid:   trigger.TriggerUID.String(),
+			Status:       modelpb.ModelTrigger_TriggerStatus(trigger.Status),
+			Visibility:   modelpb.Model_Visibility(trigger.Visibility),
+			Source:       modelpb.ModelTrigger_TriggerSource(trigger.Source),
+			StartTime:    timestamppb.New(trigger.StartTime),
+			RequesterUid: trigger.RequesterUID.String(),
```

Comment:
Requester data (also input, output data) should be anonymised if the requester != owner. 

Also, if I understood it correctly, if the context user isn't the model owner we should filter out any trigger that wasn't requested by them.

Reply:
will confirm the requirements and fix in next PR